go/vt/vttablet/tabletserver/schema/engine.go

```diff
@@ -593,6 +593,32 @@ func (se *Engine) GetSchema() map[string]*Table {
 	return tables
 }
 
+// MarshalMinimalSchema returns a protobuf encoded binlogdata.MinimalSchema
+func (se *Engine) MarshalMinimalSchema() ([]byte, error) {
+	se.mu.Lock()
+	defer se.mu.Unlock()
+	dbSchema := &binlogdatapb.MinimalSchema{
+		Tables: make([]*binlogdatapb.MinimalTable, 0, len(se.tables)),
+	}
+	for _, table := range se.tables {
+		dbSchema.Tables = append(dbSchema.Tables, newMinimalTable(table))
+	}
+	return dbSchema.MarshalVT()
```

Comment:
@frouioui thanks for backporting this! Just wanted to note that in v15 (and v16) Vitess was still using `proto.Marshal(dbSchema)` rather than `dbSchema.MarshalVT()` for this. Not sure if it's important, or why the change was only made in later versions, but wanted to call it out in case we'd prefer to continue using `proto.Marshal(dbSchema)`.

Reply:
I think this should be fine, but I will keep an eye out. Thank you for mentioning this.

@rohit-nayak-ps do you think we should revert to `proto.Marshal(dbSchema)`?

Reply:
Yes, good catch.  To be consistent, we should just use `proto.Marshal(dbSchema)` in v15/v16.